plugins/filter/resource_list_compare.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: resource_list_compare
+    author: Rohit Thakur (@rohitthakur2590)
+    version_added: "2.4.0"
+    short_description: Generate the final resource list combining/comparing base and provided resources
+    description:
+        - Generate the final list of resources after comparing with base list and provide list of resources/bangs.
+    options:
+      base:
+        description: Specify the base list of supported network resources.
+        type: list
+        elements: str
+      target:
+        description: Specify the target list of resources.
+        type: list
+        elements: str
+"""
+
+EXAMPLES = r"""
+- name: set facts for data and criteria
+  set_fact:
+    network_resources: "{{ network_resources_list['modules']|resource_list_compare(resources) }}"
+"""
+
+RETURN = """
+  list:
+    combined=['interfaces', 'l2_interfaces', 'l3_interfaces']
+"""
+
+from ansible.errors import AnsibleFilterError
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    check_argspec,
+)
+
+ARGSPEC_CONDITIONALS = {}
+
+
+def resource_list_compare(*args, **kwargs):
+    if len(args) < 2:
+        raise AnsibleFilterError(
+            "Missing either 'base' or 'other value in filter input,"
+            " refer 'ansible.network.resource_manager.resource_list_compare' filter plugin documentation for details"
+        )
+
+    params = {"base": args[0], "target": args[1]}
```

Comment:
this pattern https://github.com/ansible-collections/ansible.utils/blob/main/plugins/filter/get_path.py#L165  will let the user pass in kwargs eg something|param_list_compare(target=[1,2,3])

Reply:
Implemented. Thanks!